aspnet/security/accconfirm.rst

```diff
@@ -1,236 +1,209 @@
+
+Configure the ``AuthMessageSender`` class
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This tutorial shows how to add email notification through `SendGrid <https://sendgrid.com/>`_, but you can send email using SMTP and other mechanisms.
+ - Install the SendGrid NuGet package. From the Package Manager Console,  enter the following the following command:
+``Install-Package SendGrid``
  - Go to the `Azure SendGrid sign up page <http://azure.microsoft.com/en-us/marketplace/partners/sendgrid/sendgrid-azure/>`_ and register for a free SendGrid account.
- - Add code in **MessageServices.cs** similar to the following to configure SendGrid
+ - Add code in *Services/MessageServices.cs* similar to the following to configure SendGrid
 
-.. code-block:: c#
+.. literalinclude:: accconfirm/sample/WebApplication1/src/WebApplication1/Services/MessageServices.cs
+   :language: c#
+   :lines: 12-51
+   :dedent: 4 
+
+.. note:: SendGrid doesn't currently target dnxcore50: If you build your project you will get compilation errors. This is because SendGrid does not have a package for dnxcore50 and some APIs such as System.Mail are not available on .NET Core. You can remove dnxcore50 from *project.json* or call the REST API from SendGrid to send email. The code below shows the updated *project.json* file with ``"dnxcore50": { }`` removed.
 
-    public static Task SendEmailAsync(string email, string subject, string message)
-    {
-        // Plug in your email service here to send an email.
-        var myMessage = new SendGridMessage();
-        myMessage.AddTo(email);
-        myMessage.From = new System.Net.Mail.MailAddress("Joe@contoso.com", "Joe S.");
-        myMessage.Subject = subject;
-        myMessage.Text = message;
-        myMessage.Html = message;
-        var credentials = new NetworkCredential("SendGridUser", "SendGridKey");
-        // Create a Web transport for sending email.
-        var transportWeb = new Web(credentials);
-        // Send the email.
-        if (transportWeb != null)
-        {
-            return transportWeb.DeliverAsync(myMessage);
-        }
-        else
-        {
-            return Task.FromResult(0);
-        }
-    }
-
-.. note:: SendGrid cannot target dnxcore50: If you build your project then you will get compilation errors. This is because SendGrid does not have a package for dnxcore50 and some APIs such as System.Mail are not available on .NET Core. You can remove dnxcore50 from project.json or call the REST API from SendGrid to send email.
-
-.. note:: Security Note: Never store sensitive data in your source code. The account and credentials are added to the code above to keep the sample simple. Follow the steps on how to store secrets using the `Secret Manager <https://github.com/aspnet/Home/wiki/DNX-Secret-Configuration>`_ . The template code is setup to read configuration values from the SecretManager.
+.. code-block:: json
 
+	 "frameworks": {
+    "dnx451": { }
+  },
```

Comment:
Recommend leaving both framework lines in and highlight the one you want the user to delete. Then, tell the user to delete the highlighted line.